dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java

```diff
@@ -259,26 +259,8 @@ protected WorkflowVersion versionValidation(WorkflowVersion version, Workflow en
         version.addOrUpdateValidation(descriptorValidation);
 
         DescriptorLanguage.FileType testParameterType = null;
-        switch (identifiedType) {
-        case DOCKSTORE_SMK:
-            testParameterType = DescriptorLanguage.FileType.SMK_TEST_PARAMS;
-            break;
-        case DOCKSTORE_CWL:
-            testParameterType = DescriptorLanguage.FileType.CWL_TEST_JSON;
-            break;
-        case DOCKSTORE_WDL:
-            testParameterType = DescriptorLanguage.FileType.WDL_TEST_JSON;
-            break;
-        case DOCKSTORE_GXFORMAT2:
-            testParameterType = DescriptorLanguage.FileType.GXFORMAT2_TEST_FILE;
-            break;
-        case NEXTFLOW_CONFIG:
-            // Nextflow does not have test parameter files, so do not fail
-            break;
-        default:
-            throw new CustomWebApplicationException(identifiedType + " is not a valid workflow type.", HttpStatus.SC_BAD_REQUEST);
-        }
 
+        testParameterType = DescriptorLanguage.getDescriptorLanguage(identifiedType).getTestParamType();
```

Comment:
This is going to behave differently than the code you replaced:

1. If the identifiedType is already a test parameter file, the current code will throw an exception, your code will pass
2. If the identifiedType is not a workflow type, the current code will throw a CustomWebApplication with a contextual error message; your code will throw an UnsupportedException with a less contextual error message.

Are these intentional?